sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -276,7 +277,8 @@ class AzureSasCredentialPolicy(SansIOHTTPPolicy[HTTPRequestType, HTTPResponseTyp
 
     :param credential: The credential used to authenticate requests.
     :type credential: ~azure.core.credentials.AzureSasCredential
-    :raises: ValueError or TypeError
+    :raises ValueError: if credential is None.
+    :raises TypeError: if credential is not an instance of AzureSasCredential.
```

Comment:
It doesn't look like we raise a TypeError here or check that credential is an instance of AzureSasCredential